### PR TITLE
Feature/attempt3/upload photo

### PR DESCRIPTION
**Pull request by JinweiZhu**
Please take a look at the upload photo functionality pull request. Thanks.

### CODE REVIEW

**Comment by JinweiZhu** [comment]
The test coverage is 84 instead of the optimal 85 so CI fails. Please suggest any tests I could add though.

File: voluncheer/requirements.txt [hunk 1]
```diff
@@ -27,3 +27,4 @@ six==1.16.0
 sqlparse==0.4.3
 urllib3==1.26.14
 verify-email==2.4.3
+Pillow==9.3.0
```
Comment (joeleu) [inline voluncheer/requirements.txt:30]: I don't see this dependency in use, did I miss it or can it be removed?

**Review by joeleu — CHANGES_REQUESTED** [review]
Looking good, mostly just style requests.

File: voluncheer/requirements.txt (hunk 1 above)
Comment (JinweiZhu) [reply voluncheer/requirements.txt:30]: Pillow is required for imageField.

File: voluncheer/profiles/urls.py [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 from profiles.views.home import home
 from profiles.views.profile import ProfileView
 from profiles.views.profile import profile_update
 
 urlpatterns = [
     path("", home, name="home"),
-    path(r"^profile/$", ProfileView.as_view(), name="profile"),
-    path(r"^profile/update/$", profile_update, name="profile_update"),
-]
+    re_path(r"^profile/$", ProfileView.as_view(), name="profile"),
```
Comment (dbleek) [inline voluncheer/profiles/urls.py:12]: Thank you for fixing this. I've been meaning to get to it.

File: voluncheer/profiles/views/profile.py [hunk 3]
```diff
@@ -103,9 +103,16 @@ def password_reset_request(request):
 def profile_update(request):
     """Get profile update POST and call save function on ChangeForms."""
     if request.user.is_volunteer:
-        form = VolunteerChangeForm(request.POST, instance=request.user)
+        print("volunteer File", request.FILES)
```
Comment (dbleek) [inline voluncheer/profiles/views/profile.py:106]: Similar to Joe's other comments, do we need this print statement?

File: .github/actions/format/action.yml [hunk 4]
```diff
@@ -8,4 +8,4 @@ runs:
       run: black --check ./voluncheer
     - name: Lint Python with Flake8
       shell: bash
-      run: flake8 --filename=*.py --exclude=test_*,*_initial.py --max-line-length 100 --ignore=F401 ./voluncheer
+      run: flake8 --filename=*.py --exclude=*_initial.py --max-line-length 100 --ignore=F401 ./voluncheer
```
Comment (joeleu) [inline .github/actions/format/action.yml:11]: Can we please migrate some, if not all, of this to a `.flake8` config file in the root of the repo? We want the experience to be consistent between CI and local development.

File: voluncheer/voluncheer/urls.py [hunk 5]
```diff
@@ -42,4 +44,4 @@
     # path("chat/", include("chatroom.urls")),
     path("jobboard/", include("jobboard.urls")),
     path("map/", include("map.urls")),
-]
+] + static(settings.MEDIA_URL, document_root=settings.MEDIA_ROOT)
```
Comment (dbleek) [inline voluncheer/voluncheer/urls.py:47]: Just curious, what is this line for?

File: .gitignore [hunk 6]
```diff
@@ -7,3 +7,7 @@ __pycache__/
 voluncheer/db.sqlite3
 **/.coverage
 **/coverage.lcov
+.vscode/
```
Comment (joeleu) [inline .gitignore:10]: Please don't exclude the `.vscode` directory, we should keep the format and lint settings the same across our environments.

**Review by dbleek — CHANGES_REQUESTED** [review]
Looks good, a couple of minor changes. Please see my comment in test_forms.py.

File: voluncheer/profiles/forms/organizations.py [hunk 7]
```diff
@@ -33,13 +37,19 @@ class OrganizationChangeForm(UserChangeForm):
...
     def save(self, commit=True):
         user = self.instance
         organization = Organization.objects.get(pk=user)
         if self.is_valid():
             organization.name = self.cleaned_data.get("name")
+            organization.photo = self.cleaned_data.get("photo")
             organization.save()
+        else:
+            logger = logging.getLogger(__name__)
+            logger.exception(self.errors)
```
Comment (joeleu) [inline voluncheer/profiles/forms/organizations.py:55]: logger.exception is only to be used in an exception handler: https://docs.python.org/3/library/logging.html#logging.Logger.exception

logger.error or logger.warning would be more appropriate.

File: voluncheer/profiles/forms/volunteers.py [hunk 8]
```diff
@@ -46,20 +52,22 @@ def save(self, commit=True):
...
+
         if self.is_valid():
             volunteer.first_name = self.cleaned_data.get("first_name")
             volunteer.last_name = self.cleaned_data.get("last_name")
             volunteer.date_of_birth = self.cleaned_data.get("date_of_birth")
+            volunteer.photo = self.cleaned_data.get("photo")
             volunteer.save()
+        else:
+            logger = logging.getLogger(__name__)
+            logger.exception(self.errors)
```
Comment (joeleu) [inline voluncheer/profiles/forms/volunteers.py:73]: Same as the other file.

File: voluncheer/profiles/views/profile.py (hunk 3 above)
Comment (joeleu) [inline voluncheer/profiles/views/profile.py:106]: Looks like this print statement is left over from testing?

**Review by joeleu — CHANGES_REQUESTED** [review]

File: .github/actions/format/action.yml (hunk 4 above)
Comment (JinweiZhu) [reply .github/actions/format/action.yml:11]: I thought you don't want test_* files in CI? I removed the .flake8 because it doesn't seem useful.

Comment (JinweiZhu) [reply .github/actions/format/action.yml:11]: Or do you want me to add a .flake8 and push?

File: voluncheer/voluncheer/urls.py (hunk 5 above)
Comment (JinweiZhu) [reply voluncheer/voluncheer/urls.py:47]: It allows you to access image in media folder via url when you are in static/local debugging environment.

**Pull request merged**